### ISSUE TITLE
conda info fails on assert if CONDA_BLD_PATH is defined

### ISSUE THREAD

**Issue by msarahan** [conda/conda#3741]
@afeblot commented on [Wed Oct 26 2016](https://github.com/conda/conda-build/issues/1485)

Is it forbidden to run conda build with CONDA_BLD_PATH defined, while using the root env?
```
    Traceback (most recent call last):
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/exceptions.py", line 479, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/cli/main.py", line 145, in _main
        exit_code = args.func(args, p)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/cli/main_info.py", line 225, in execute
        channels = list(prioritize_channels(channels).keys())
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/models/channel.py", line 380, in prioritize_channels
        channel = Channel(chn)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/models/channel.py", line 161, in __call__
        c = Channel.from_value(value)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/models/channel.py", line 221, in from_value
        if value in context.custom_multichannels:
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/_vendor/auxlib/decorators.py", line 265, in new_fget
        cache[inner_attname] = func(self)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/base/context.py", line 254, in custom_multichannels
        'local': (self.local_build_root_channel,),
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/_vendor/auxlib/decorators.py", line 265, in new_fget
        cache[inner_attname] = func(self)
      File "/raid1/condapoc/root/lib/python2.7/site-packages/conda/base/context.py", line 246, in local_build_root_channel
        assert name == 'conda-bld'
    AssertionError
```

---

@msarahan commented on [Wed Oct 26 2016](https://github.com/conda/conda-build/issues/1485#issuecomment-256528953)

Thanks for reporting.  This seems like a bug in conda, not conda-build.  It appears to me that it has a hard-coded assumption that the local build root channel will always be named conda-bld, which is not a valid assumption.  I'm going to move this issue to the conda issue tracker.

**Comment by msarahan** [comment]
https://github.com/conda/conda/blob/4f6ab97f5104b6bfeb667eeb1586e56f29ec2994/conda/base/context.py#L114-L125

**Comment by kalefranz** [comment]
Also related to https://github.com/conda/conda/issues/3732

**Comment by kalefranz** [comment]
I talked through the changes in 4.2.10/4.2.11 with @mcg1969.  That is, a channel must have an identifiable name (separate from its location) if we are to make the concept of channel a first-class citizen in a package specification alongside package name and version.

Actually, I think I can still patch this one though.  The name for the conda-build channel is 'local', and I've written that into the code (with test).  I just need to fully support the dynamic nature of the `CONDA_BLD_PATH` env var.

**Comment by kalefranz** [comment]
Also, as I said in https://github.com/conda/conda/issues/3732#issuecomment-256467515, an assertion error is *always* a bug.

Whether it's a logic bug or a type bug or a bug where we just need to give a better user message, it's nevertheless a bug.

**Comment by kalefranz** [comment]
resolved with #3747
